### PR TITLE
feat: accept import.js as Base64 encoded string to /tools/import endpoint

### PR DESCRIPTION
**Pull request by swetabar**
## Related Issues
[SITES-22752](https://jira.corp.adobe.com/browse/SITES-22752)

API spec: https://opensource.adobe.com/spacecat-api-service/#tag/import/operation/startImportJob 

Support the importScript request body property. Decode the payload and write it as a file to the S3 bucket at import/{jobId}/import.js.

We use the `maxLengthImportScript` (environment variable) to set a limit on the size of the importScript.

### CODE REVIEW

File: src/support/import-supervisor.js [hunk 1]
```diff
@@ -142,14 +145,40 @@ function ImportSupervisor(services, config) {
...
+
+    // Check for the length of the importScript
+    if (importScript.length > maxLengthImportScript) {
+      throw new ErrorWithStatusCode(`Bad Request: importScript should be less than ${maxLengthImportScript} characters`, 400);
+    }
+
+    let decodedScript;
+    try {
+      decodedScript = Buffer.from(importScript, 'base64').toString('utf-8');
+    } catch (error) {
```
Comment (solaris007) [inline src/support/import-supervisor.js:161]: it seems you're not making use of the `error` param, as such you can use `} catch {` directly

File: src/support/import-supervisor.js [hunk 2]
```diff
@@ -142,14 +145,40 @@ function ImportSupervisor(services, config) {
...
+      throw new ErrorWithStatusCode('Bad Request: importScript should be a base64 encoded string', 400);
+    }
+
+    const key = `import/${jobId}/import.js`;
+    const command = new PutObjectCommand({ Bucket: s3Bucket, Key: key, Body: decodedScript });
+    try {
+      await s3Client.send(command);
+    } catch (error) {
```
Comment (solaris007) [inline src/support/import-supervisor.js:169]: it seems you're not making use of the `error` param, as such you can use `} catch {` directly

**Review by solaris007 — APPROVED** [review]
LGTM, just minor comment

**Review by blefebvre — APPROVED** [review]
Looks good! Just a couple of minor items.

File: src/support/import-supervisor.js [hunk 3]
```diff
@@ -162,6 +191,9 @@ function ImportSupervisor(services, config) {
 
     // Custom import.js scripts are not initially supported.
     // Future: Write import.js to the S3 bucket, at {S3_BUCKET_NAME}/import/{jobId}/import.js
+    if (importScript) {
```
[the remark this answers is not in the document]
Comment (swetabar) [reply src/support/import-supervisor.js:194]: I'm checking it later in the function: `writeImportScriptToS3` and I throw an error there if `importScript` is not a string.

**Pull request merged**